core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowResource.scala

```diff
@@ -443,4 +444,107 @@ class WorkflowResource {
       workflowDao.update(userWorkflow)
     }
   }
+
+  /**
+    * This method performs a full-text search in the content column of the
+    * workflow table for workflows that match the specified keywords.
+    *
+    * This method utilizes MySQL Boolean Full-Text Searches
+    * reference: https://dev.mysql.com/doc/refman/8.0/en/fulltext-boolean.html
+    * @param sessionUser The authenticated user.
+    * @param keywords    The search keywords.
+    * @return A list of workflows that match the search term.
+    */
+  @GET
+  @Path("/search")
+  def searchWorkflows(
+      @Auth sessionUser: SessionUser,
+      @QueryParam("query") keywords: java.util.List[String]
+  ): List[DashboardWorkflowEntry] = {
+    val user = sessionUser.getUser
+    if (keywords.size() == 0) {
+      return List.empty[DashboardWorkflowEntry]
+    }
+    // make sure keywords don't end with "+-()<>~*\"", these are reserved for SQL full-text boolean operator
+    val forbiddenChars: Set[Char] = "+-()<>~*\"".toSet
+    val modifiedKeywords: Iterable[String] = keywords.map { keyword =>
+      // remove all reserved characters in keyword
+      keyword.filterNot(forbiddenChars.contains)
+    }
+    var matchQuery: Condition = noCondition()
+    for (key: String <- modifiedKeywords) {
+      val words = key.split("\\s+")
+      if (words.length > 1 || key.contains('@')) {
```

Comment:
I added clarification commments to this. Because the case where sub-string search is the "normal" case, it should go first.
Handle `words.length <= 1` seperately. (not included in the code below)
````
      if (!key.contains('@')) {
        // Use '*' to enable substring search.
        matchQuery = matchQuery.and(
          "(MATCH(texera_db.workflow.name, texera_db.workflow.description, texera_db.workflow.content) AGAINST(+{0}'*' IN BOOLEAN mode) OR " +
            "MATCH(texera_db.user.name) AGAINST (+{0}'*' IN BOOLEAN mode) " +
            "OR MATCH(texera_db.project.name, texera_db.project.description) AGAINST (+{0}'*' IN BOOLEAN mode))",
          key
        )
      } else {
        // When the search query contains "@" character, it conflicts with the @distance operator. In this case, sub-string search cannot be supported.
        matchQuery = matchQuery.and(
          "(MATCH(texera_db.workflow.name, texera_db.workflow.description, texera_db.workflow.content) AGAINST(+{0} IN BOOLEAN mode) OR " +
            "MATCH(texera_db.user.name) AGAINST (+{0} IN BOOLEAN mode) " +
            "OR MATCH(texera_db.project.name, texera_db.project.description) AGAINST (+{0} IN BOOLEAN mode))",
          '"' + key + '"'
        )
      }
````

Reply:
if we decide use replace reserved characters with space, then there is no need to handle  `if (!key.contains('@'))` this case anymore